python/sglang/srt/utils/common.py

```diff
@@ -162,6 +162,22 @@ def _check(cc_major):
     ) >= (12, 3)
 
 
+@contextmanager
+def device_context(device: torch.device):
+    if device.type == "cuda" and is_cuda():
```

Comment:
The origin implement is `torch.get_device_module(x.device).device(x.device.index)`
And this implement will make other vendor unavaliable and not following the registry style.
Are there any possibility to make the code like below?
```python
    module = torch.get_device_module(x.device)
    if module is not None:
        with module.device(device.index):
                yield
    else:
        # raise some exception
```